### PR TITLE
R-UST rises again.

### PR DESCRIPTION
**Pull request by Mechoid**
:cl:
fix - The R-UST's magnetic traps now function properly.
fix - The R-UST now uses a more sane method of tracking fuel. It is now measured in ten-thousandths of a reagent unit.
tweak - The R-UST now uses a crude TGUI. Interface design is my passion.
rscadd - The R-UST can now be used to properly create materials through the iron-iron reaction. Retrieving them is only possible via field shutdown.
rscadd - The R-UST can syphon reagents from the field into its internal storage, accessible only via hose.
/:cl:

Fixes https://github.com/PolarisSS13/Polaris/issues/7829 , Fixes https://github.com/PolarisSS13/Polaris/issues/7830 , Fixes https://github.com/PolarisSS13/Polaris/issues/7828 , Fixes https://github.com/PolarisSS13/Polaris/issues/7831 , Fixes https://github.com/PolarisSS13/Polaris/issues/7832

Somewhat implements https://github.com/PolarisSS13/Polaris/issues/7827 

image

### CODE REVIEW

**Comment by Atermonera** [comment]
Pics of the UI? And if this actually fixes any of the R-UST issues that I tasked you with please include those in the PR desc so they can be properly closed.

**Comment by Shadow-Quill** [comment]
As a note, you have to put "Fixes #XXXX, Fixes #XXXX, etc" in order to actually close each issue.

File: code/modules/tgui/modules/rustcore_monitor.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		return TRUE
+
+	for(var/parameter in params)
+		to_world("[parameter] - [params[parameter]]")
+
+	switch(action)
+		if("toggle_active")
+			var/obj/machinery/power/fusion_core/C = locate(params["core"])
+			if(!istype(C))
+				return 0
```
Comment (Atermonera) [inline code/modules/tgui/modules/rustcore_monitor.dm:18]: TRUE/FALSE

File: code/modules/tgui/modules/rustfuel_control.dm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		return TRUE
+
+	for(var/parameter in params)
+		to_world("[parameter] - [params[parameter]]")
+
+	switch(action)
+		if("toggle_active")
+			var/obj/machinery/fusion_fuel_injector/FI = locate(params["fuel"])
+			if(!istype(FI))
+				return 0
```
Comment (Atermonera) [inline code/modules/tgui/modules/rustfuel_control.dm:18]: TRUE/FALSE

File: code/modules/power/fusion/core/core_control.dm [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
 
+/obj/machinery/computer/fusion_core_control/attack_hand(var/mob/user as mob)
+	..()
+	if(stat & (BROKEN|NOPOWER))
+		return
+
+	monitor.tgui_interact(user)
+
+/*
 /obj/machinery/computer/fusion_core_control/attack_hand(mob/user)
```
[the remark this answers is not in the document]
Comment (Mechoid) [reply code/modules/power/fusion/core/core_control.dm:44]: Commented out in order to keep for reference, since the TGUI interaction is far simpler.

**Pull request merged**